packages/react-components/react-alert/src/components/Alert/useAlert.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import { Button } from '@fluentui/react-button';
+import {
+  CheckmarkCircle16Filled,
+  DismissCircle16Filled,
+  ErrorCircle16Filled,
+  Warning16Filled,
+} from '@fluentui/react-icons';
+import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
+
+import type { AlertProps, AlertState } from './Alert.types';
+
+/**
+ * Create the state required to render Alert.
+ *
+ * The returned state can be modified with hooks such as useAlertStyles_unstable,
+ * before being passed to renderAlert_unstable.
+ *
+ * @param props - props from this instance of Alert
+ * @param ref - reference to root HTMLElement of Alert
+ */
+export const useAlert_unstable = (props: AlertProps, ref: React.Ref<HTMLElement>): AlertState => {
+  const { intent } = props;
+
+  /** Determine the icon to render based on the intent */
+  let iconToUse;
+  switch (intent) {
+    case 'success':
+      iconToUse = <CheckmarkCircle16Filled />;
+      break;
+    case 'error':
+      iconToUse = <DismissCircle16Filled />;
+      break;
+    case 'warning':
+      iconToUse = <Warning16Filled />;
+      break;
+    case 'info':
+      iconToUse = <ErrorCircle16Filled />;
+      break;
+  }
+
+  const icon = resolveShorthand(props.icon, {
+    defaultProps: {
+      children: iconToUse,
+    },
+    required: true,
+  });
+
+  return {
+    components: {
+      root: 'div',
+      icon: 'span',
+      action: Button,
+    },
+    root: getNativeElementProps('div', {
+      ref,
+      children: props.children,
+      ...props,
+    }),
+    icon: icon,
+    action: resolveShorthand(props.action),
+    intent,
+    isIntentIcon: !!props.intent && !props.icon,
```

Comment:
Can you please clarify why its needed? As I see it applies a classname that is used only in tests...

Reply:
Yeah, I added it specifically to determine if the rendered icon is due to icon or intent prop. I use this in tests to determine the expected behavior when both icon & intent prop are provided.

Reply:
I also need it in styles to determine the color for the icon

Reply:
- I don't think that it's a good approach to ship code that is needed only for tests to customers. They will get penalties for nothing.
  - That test is not really useful as it tests that a `className` is applied, nothing more.
  - _Suggestion:_ Delete that test & `className`, use visual tests for these checks as the source of truth
- I still don't understand why `isIntentIcon` is needed to apply styles. If you don't want to get styles for `intent` do not use `intent` prop. I mean this case:
  ```tsx
  <Alert intent="success" icon={<DeleteRegular />} />
  ```
  For me that behavior looks more confusing than correct as then I can get colored icons only with custom styles.

Reply:
Updated